src/rt/sched/behaviourcore.h

```diff
@@ -50,7 +61,11 @@ namespace verona::rt
   {
     Cown* cown;
     /**
-     * Possible vales:
+     * Possible vales before scheduling:
+     *   0 - NoTransfer
+     *   1 - YesTransfer
```

Comment:
I think there is an issue here.  The `YesTransfer` value could be read as `Ready` by a subsequently enqueued behaviour before this has been linked in.

Consider
```
when(std::move(c1a)) { B1 } || when (c1b) { B2 }
``` 
So the following sequence seems problematic
1. Exchange B1 on Cown c1 with YesTransfer
2. Exchange B2 on Cown c1 with NoTransfer 
3. B2 looks at the request it got from exchange, and observes YesTransfer, but views it as Ready.
4. B2 overwrites YesTransfer with a pointer to this behaviour
5. B1 calls `reset_yes_transfer` and overwrites the pointer to B2.

Now a behaviour is lost, and a when B1 completes, the scheduler thread will spin forever waiting for the pointer to be set.

It would be great if systematic testing covered this case and failed, before you fix the representation.

Reply:
The code is right, I am wrong.